### PR TITLE
add CI

### PR DESCRIPTION
**Pull request by Borda**
adding Continues Integration

### CODE REVIEW

**Comment by cclauss** [comment]
Please consider adding [flake8](http://flake8.pycqa.org/en/latest/) testing to this PR because this repo has seven undefined names...

flake8 testing of https://github.com/aleju/imgaug on Python 2.7.14

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./imgaug/imgaug.py:304:13: F821 undefined name 'ia'
    return [ia.new_random_state(seed+i) for i in sm.xrange(n)]
            ^
./imgaug/parameters.py:138:5: F821 undefined name 'misc'
    misc.imshow(
    ^
./imgaug/parameters.py:2503:29: F821 undefined name 'sigmoid_thresh'
        elif ia.is_iterable(sigmoid_thresh):
                            ^
./imgaug/augmenters/color.py:289:25: F821 undefined name 'p'
        elif isinstance(p, StochasticParameter):
                        ^
./imgaug/augmenters/convolutional.py:118:118: F821 undefined name 'sigma'
            raise Exception("Expected float, int, tuple/list with 2 entries or StochasticParameter. Got %s." % (type(sigma),))
                                                                                                                     ^
./imgaug/augmenters/meta.py:1031:41: F821 undefined name 'global_error_msg'
                        raise Exception(global_error_msg)
                                        ^
./imgaug/augmenters/meta.py:1046:37: F821 undefined name 'global_error_msg'
                    raise Exception(global_error_msg)
                                    ^
```

**Comment by Borda** [comment]
@cclauss  I feel confused because what you show are not my changes...

**Comment by aleju** [comment]
Sorry for the late reply, was quite busy recently.
I made some changes to your branch to resolve conflicts with recent updates to `test.py`. The PR can be merged once they are added.

**Comment by Borda** [comment]
@aleju  I just resolved conflicts in tests...

File: .travis.yml [hunk 1]
```diff
@@ -2,22 +2,43 @@ sudo: required
...
-  - cp $(python -c 'import cv2; print(cv2.__file__)') venv/lib/python2.7/site-packages
-  - . venv/bin/activate
-  - pip install -U numpy  # for copied cv2
+#  - virtualenv venv
+#  - . venv/bin/activate
+  - pip install -r requirements.txt
+  - pip install coverage codecov pytest>=3.0.5 pytest-cov codacy-coverage # nose>=1.3.7
   - pip install .
 
 script:
```
Comment (cclauss) [inline .travis.yml]:
```
before_script:
  - pip install coveralls flake8
  # stop the build if there are Python syntax errors or undefined names
  - flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics
  # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
  - flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

**Comment by aleju** [comment]
Thanks for the help! This is now in master. The flake8 test for errors was also added and related errors were fixed. The flake8 test for code style was not yet added, as the project has way too many style problems for that (especially regarding line widths).

**Comment by cclauss** [comment]
Makes sense.  __autopep8__ or __yapf__ can auto-fix most style issues but they are clearly optional.

**Pull request merged**